src/v/cluster/metadata_dissemination_handler.cc

```diff
@@ -85,7 +85,8 @@ metadata_dissemination_handler::do_update_leadership(
 
 static get_leadership_reply
 make_get_leadership_reply(const partition_leaders_table& leaders) {
-    std::vector<ntp_leader> ret;
+    fragmented_vector<ntp_leader> ret;
```

Comment:
Maybe explain in the commit message what prompted us to switch to `fragmented_vector` in this response type in the first place.